### PR TITLE
[Libs] Deprecated routerDialog moved changes to ActivityHandlerDialog

### LINKED ISSUE #2584: Router dialog -"complete" message shown in response to events

### PR DESCRIPTION
**Pull request by lauren-mills**
Close #2584 

### Purpose
*What is the context of this pull request? Why is it being done?*
This reverts the changes to routerDialog and adds the ActivityHandlerDialog instead. It deprecates the old routerDialog with a note for people to migrate.

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
- Once this is merged, I'll update VA sample with the new lib

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+﻿using System;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Solutions.Extensions;
 using Microsoft.Bot.Schema;
 
 namespace Microsoft.Bot.Builder.Solutions.Dialogs
 {
+    [Obsolete("RouterDialog has been replaced by ActivityHandlerDialog. RouterDialog will no longer be supported.", false)]
```
Comment (darrenj) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs]: Can you adjust to `being replaced` and "no longer be supported moving forward" and add a aka.ms link to a doc page?

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/ActivityHandlerDialog.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            : base(dialogId, telemetryClient)
+        {
+            TelemetryClient = telemetryClient;
+        }
+
+        protected override Task<DialogTurnResult> OnBeginDialogAsync(DialogContext innerDc, object options, CancellationToken cancellationToken = default)
+        {
+            return OnContinueDialogAsync(innerDc, cancellationToken);
+        }
+
```
Comment (darrenj) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/ActivityHandlerDialog.cs:40]: doc elements needed as per other methods (do across them all)

**Review by darrenj — CHANGES_REQUESTED** [review]

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs (hunk 1 above)
Comment (ryanisgrig) [reply sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs]: Or just `is replaced`

Comment (lauren-mills) [reply sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs]: How is this?

image

Comment (ryanisgrig) [reply sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs]: Looks good to me

**Review by ryanisgrig — APPROVED** [review]

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**